TestDWZCompression fails on linux/loong64

On linux/loong64, the test case TestDWZCompression is currently skipped instead of passing. When it is not skipped, it runs but fails.

Run the following command:
```
GOFLAGS="-tags=exp.linuxloong64" go test -v -count=1 -run=TestDWZCompression$ ./pkg/proc
```

Output when not skipped:
```
=== RUN   TestDWZCompression
    proc_test.go:3655: proc_test.go:3655: EvalVariable("stdin"): could not find symbol value for stdin
--- FAIL: TestDWZCompression (0.71s)
FAIL
FAIL	github.com/go-delve/delve/pkg/proc	0.716s
```

OS/Arch: linux/loong64
Go version: go version go1.24.3 linux/loong64

--- a/Documentation/backend_test_health.md
+++ b/Documentation/backend_test_health.md
@@ -35,8 +35,7 @@ Tests skipped by each supported backend:
 * linux/riscv64 skipped = 2
 	* 1 broken - cgo stacktraces
 	* 1 not working on linux/riscv64
-* loong64 skipped = 8
-	* 1 broken
+* loong64 skipped = 7
 	* 1 broken - global variable symbolication
 	* 6 not implemented
 * pie skipped = 2

--- a/pkg/proc/loong64_arch.go
+++ b/pkg/proc/loong64_arch.go
@@ -45,7 +45,10 @@ func LOONG64Arch(goos string) *Arch {
 
 func loong64FixFrameUnwindContext(fctxt *frame.FrameContext, pc uint64, bi *BinaryInfo) *frame.FrameContext {
 	a := bi.Arch
-	if fctxt == nil {
+	if a.sigreturnfn == nil {
+		a.sigreturnfn = bi.lookupOneFunc("runtime.sigreturn")
+	}
+	if fctxt == nil || (a.sigreturnfn != nil && pc >= a.sigreturnfn.Entry && pc < a.sigreturnfn.End) {
 		// When there's no frame descriptor entry use BP (the frame pointer) instead
 		// - return register is [bp + a.PtrSize()] (i.e. [cfa-a.PtrSize()])
 		// - cfa is bp + a.PtrSize()*2
@@ -75,6 +78,26 @@ func loong64FixFrameUnwindContext(fctxt *frame.FrameContext, pc uint64, bi *Bina
 				Reg:    regnum.LOONG64_FP,
 				Offset: int64(2 * a.PtrSize()),
 			},
+		}
+	}
+
+	if a.crosscall2fn == nil {
+		a.crosscall2fn = bi.lookupOneFunc("crosscall2")
+	}
+
+	if a.crosscall2fn != nil && pc >= a.crosscall2fn.Entry && pc < a.crosscall2fn.End {
+		rule := fctxt.CFA
+		if rule.Offset == crosscall2SPOffsetBad {
+			rule.Offset += crosscall2SPOffset
+		}
+		fctxt.CFA = rule
+	}
+
+	if fctxt.CFA.Rule == frame.RuleUndefined {
+		fctxt.CFA = frame.DWRule{
+			Rule:   frame.RuleCFA,
+			Reg:    regnum.LOONG64_FP,
+			Offset: int64(2 * a.PtrSize()),
 		}
 	}
 

--- a/pkg/proc/proc_test.go
+++ b/pkg/proc/proc_test.go
@@ -3644,7 +3644,6 @@ func TestIssue951(t *testing.T) {
 func TestDWZCompression(t *testing.T) {
 	skipOn(t, "broken", "ppc64le")
 	skipOn(t, "broken", "riscv64")
-	skipOn(t, "broken", "loong64")
 	// If dwz is not available in the system, skip this test
 	if _, err := exec.LookPath("dwz"); err != nil {
 		t.Skip("dwz not installed")

